afs/afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageDataSource.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.storage;
+
+import com.powsybl.commons.datasource.DataSource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class AppStorageDataSource implements DataSource {
+
+    private static final String SEPARATOR = "__";
+
+    public interface Name {
+
+        static Name parse(String text) {
+            Objects.requireNonNull(text);
+            if (text.startsWith(SuffixAndExtension.START_PATTERN)) {
+                int pos2 = text.indexOf(SEPARATOR, SuffixAndExtension.START_PATTERN.length());
+                if (pos2 == -1) {
+                    throw new IllegalStateException("Second separator not found");
+                }
+                String suffixe = text.substring(SuffixAndExtension.START_PATTERN.length(), pos2);
+                String ext = text.substring(pos2 + SEPARATOR.length());
+                return new SuffixAndExtension(suffixe, ext);
+            } else if (text.startsWith(FileName.START_PATTERN)) {
+                String fileName = text.substring(FileName.START_PATTERN.length());
+                return new FileName(fileName);
+            } else {
+                return null;
+            }
+        }
+    }
+
+    public static class SuffixAndExtension implements Name {
+
+        public static final String START_PATTERN = "DATA_SOURCE_SUFFIX_EXT" + SEPARATOR;
+
+        private final String suffix;
+
+        private final String ext;
+
+        SuffixAndExtension(String suffix, String ext) {
+            this.suffix = suffix;
+            this.ext = ext;
+        }
+
+        public String getSuffix() {
+            return suffix;
+        }
+
+        public String getExt() {
+            return ext;
+        }
+
+        @Override
+        public String toString() {
+            return START_PATTERN + Objects.toString(suffix, "") + SEPARATOR + Objects.toString(ext, "");
+        }
+    }
+
+    public static class FileName implements Name {
+
+        public static final String START_PATTERN = "DATA_SOURCE_FILE_NAME" + SEPARATOR;
+
+        private final String name;
+
+        public FileName(String name) {
```

Comment:
This constructor is public but SuffixAndExtension constructor is not.

Reply:
Changed to private package